src/hybrid/parse/hybrid_callbacks.c

```diff
@@ -0,0 +1,404 @@
+    }
+}
+
+// LIMIT callback - implements EXACT original logic from lines 259-296
+void handleLimit(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ArgsCursor *ac = (ArgsCursor*)value;
+    QueryError *status = ctx->status;
+    ctx->specifiedArgs |= SPECIFIED_ARG_LIMIT;
+
+    if (AC_NumRemaining(ac) < 2) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "LIMIT requires 2 arguments");
+        return;
+    }
+
+    long long offset, num;
+    if (AC_GetLongLong(ac, &offset, AC_F_GE0) != AC_OK) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "LIMIT offset must be a non-negative integer");
+        return;
+    }
+    if (AC_GetLongLong(ac, &num, AC_F_GE0) != AC_OK) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "LIMIT count must be a non-negative integer");
+        return;
+    }
+
+    if (num == 0 && offset != 0) {
+        QueryError_SetError(status, QUERY_ELIMIT, "The `offset` of the LIMIT must be 0 when `num` is 0");
+        return;
+    }
+
+    if (num == 0) {
+        // LIMIT 0 0 - only count
+        REQFLAGS_AddFlags(ctx->reqFlags, QEXEC_F_NOROWS);
+        REQFLAGS_AddFlags(ctx->reqFlags, QEXEC_F_SEND_NOFIELDS);
+        // TODO: unify if when req holds only maxResults according to the query type.
+        //(SEARCH / AGGREGATE)
+    } else if (num > *ctx->maxResults) {
+        QueryError_SetWithoutUserDataFmt(status, QUERY_ELIMIT, "LIMIT exceeds maximum of %llu",
+                             *ctx->maxResults);
+        return;
+    } else if (offset > LLONG_MAX - num) {
+        QueryError_SetError(status, QUERY_EPARSEARGS, "LIMIT offset + count overflow");
+        return;
+    }
+
+    PLN_ArrangeStep *arng = AGPLN_GetOrCreateArrangeStep(ctx->plan);
+    arng->isLimited = 1;
+    arng->offset = offset;
+    arng->limit = num;
+}
+
+// SORTBY callback - implements EXACT original logic from lines 298-323
+void handleSortBy(ArgParser *parser, const void *value, void *user_data) {
+    HybridParseContext *ctx = (HybridParseContext*)user_data;
+    ArgsCursor *ac = (ArgsCursor*)value;
+    QueryError *status = ctx->status;
+    ctx->specifiedArgs |= SPECIFIED_ARG_SORTBY;
+
+    // We managed to get a valid arg cursor
+    if (AC_IsAtEnd(ac)) {
+        // We essentially got SORTBY 0
+        *ctx->reqFlags |= QEXEC_F_NO_SORT;
+        return;
+    }
+
+    PLN_ArrangeStep *arng = AGPLN_GetOrCreateArrangeStep(ctx->plan);
+    // Parse field/direction pairs
+    while (!AC_IsAtEnd(ac)) {
+        const char *field = AC_GetStringNC(ac, NULL);
+        if (!field) {
+            QueryError_SetError(status, QUERY_EPARSEARGS, "Missing field name in SORTBY");
+            return;
+        }
+
+        // Remove '@' prefix if present (same logic as parseSortby)
+        if (*field == '@') {
+            field++;  // Skip the '@' prefix
+        }
+
+        // Default to ascending
+        int ascending = 1;
```

Comment:
it may be nice to have some definition "DEFAULT_SORTBY_BEHAVIOR" and ASCENDING as 1 and DESC as 0.